traitlets/traitlets.py

```diff
@@ -1199,7 +1327,7 @@ def on_trait_change(self, handler=None, name=None, remove=False):
         else:
             self.observe(_callback_wrapper(handler), names=name)
 
-    def observe(self, handler, names=All, type='change'):
+    def observe(self, handler, names=None, tags=None, type='change'):
```

Comment:
Yeah, I'd get in the habit of putting new args at the end. Most of these should be treated (in documentation, etc.) as keyword-only, even if they *can* be positional.